### PR TITLE
[Merged by Bors] - feat(AlgebraicGeometry/Morphisms): define closed immersions

### PR DESCRIPTION
**Pull request by chrisflav**
Defines closed immersions of schemes.

Co-authored-by: Amelia Livingston @101damnations
Co-authored-by: Jonas van der Schaaf @jonasvanderschaaf

---

This is a reopening and clean up of #6135. The original author is Jonas van der Schaaf who no longer wanted to pursue this PR.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by erdOne** [comment]
Thanks! I have some thoughts on how we should define closed immersions, but they require a substantial more work and should not block this. We can refactor this later when the time comes.
maintainer merge

File: Mathlib/AlgebraicGeometry/Morphisms/ClosedImmersion.lean [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+* `IsClosedImmersion` : The property of scheme morphisms stating `f : X ⟶ Y` is a closed immersion.
+
+## TODO
+
+* Show closed immersions of affines are induced by surjective ring maps
+* Show closed immersions are stable under pullback
+* Define closed immersions of locally ringed spaces, where we also assume that the kernel of `O_X →
+  f_*O_Y` is locally generated by sections as an `O_X`-module, and relate it to this file. See
+  https://stacks.math.columbia.edu/tag/01HJ.
```
Comment (kbuzzard) [inline Mathlib/AlgebraicGeometry/Morphisms/ClosedImmersion.lean:27]: Other TODOs: one of my favourite theorems about closed immersions is that they're precisely the proper monomorphisms. One could imagine doing the easy direction :-)

**Comment by kbuzzard** [comment]
Thanks a lot!

bors d+

**Pull request closed without merging**